sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -448,6 +433,145 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       "ALTER TABLE unset_prop_table UNSET TBLPROPERTIES ('prop1')")
   }
 
+  /**
+   * To see if stats exist, we need to check spark's stats properties instead of catalog
+   * statistics, because hive would change stats in metastore and thus change catalog statistics.
```

Comment:
Hive updates stats for some formats, e.g. in the test `test statistics of LogicalRelation converted from Hive serde tables`, hive updates stats for orc table but not parquet table.
I think it's tricky and fussy to relay on hive's stats, we should rely on spark's stats. Here I just eliminate hive's influence for checking. As in `CatalogStatistics` we first fill in hive's stats and then override it using spark's stats, we can't tell where the stats is from if we check `CatalogStatistics`.